Framework/PythonInterface/plugins/algorithms/RefineSatellitePeaks.py

```diff
@@ -0,0 +1,181 @@
+from mantid.kernel import *
+from mantid.api import *
+from mantid.simpleapi import *
+
+import numpy as np
+from scipy.spatial import KDTree
+from scipy.cluster.vq import kmeans2
+import scipy.cluster.hierarchy as hcluster
+
+
+class RefineSatellitePeaks(DataProcessorAlgorithm):
+
+    def category(self):
+        return 'Crystal\\Peaks'
+
+    def PyInit(self):
+        self.declareProperty(IPeaksWorkspaceProperty(name="MainPeaks",
+                                                     defaultValue="",
+                                                     direction=Direction.Input),
+                             doc="Main integer HKL peaks. Q vectors will be calculated relative to these peaks.")
+
+        self.declareProperty(IPeaksWorkspaceProperty(name="SatellitePeaks",
+                                                     defaultValue="",
+                                                     direction=Direction.Input),
+                             doc="Positions of seed satellite peaks. These will be used to define the q vectors for each satellite.")
+
+        self.declareProperty(WorkspaceProperty(name="MDWorkspace",
```

Comment:
But not just any MDWorkspace, must be in the HKL frame. Maybe add to docs for property.

Reply:
Really it should be validated with a property validator as well. I don't think we currently have a validator for MD space type do we?